catalyst_voices/packages/libs/catalyst_cardano_serialization/lib/src/address.dart

```diff
@@ -115,7 +115,7 @@ class ShelleyAddress extends Equatable implements CborEncodable {
   }
 
   @override
-  int get hashCode => Object.hash(bytes, hrp);
+  int get hashCode => toBech32().hashCode;
```

Comment:
The `hashCode` has to be in sync with `equals` operator, is there any reason that `hashCode` uses `bech32` and equals uses both `bytes` and `hrp`?

Reply:
The bytes is a List, meaning it would generate different hashCode for any new address instance.

Reply:
Alright, I see. Though I think this solution is not ideal because now we aren't comparing apples to apples but rather using bananas in `hashCode` and apples in `equals`.

Instead of hashing the list itself we should be able to obtain the hash of all items in the list, like so:

```suggestion
  int get hashCode => Object.hashAll([...bytes, hrp]);
```

This should solve the problem when using different List implementations and we'd retain the symmetry between `hashCode` and `equals`

Reply:
yeah it makes sense